src/init.cpp

```diff
@@ -194,6 +230,7 @@ void Interrupt()
 void Shutdown(InitInterfaces& interfaces)
 {
     LogPrintf("%s: In progress...\n", __func__);
+    RemoveFile(GetReadyFile());
```

Comment:
If the computer crashes, the file won't get deleted...

Reply:
Fair point, but the external command approach would likely have the same limitation. Perhaps we could remove any existing `ready` file early in `AppInitMain()` (or earlier) to mitigate this?

I am very uncomfortable with adding the ability to `exec()` external programs. It just feels like an attack vector.

Reply:
`-blocknotify` and `-walletnotify` already exist.